crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -361,25 +354,25 @@ impl AlgorithmUpdaterV1 {
             .saturating_add(projection_portion);
     }
 
+    fn descaled_exec_price(&self) -> u64 {
+        let factor = self.gas_price_factor.into();
+        self.new_scaled_exec_price
+            .checked_div(factor)
+            .expect("factor is never 0")
+    }
+
+    fn descaled_da_price(&self) -> u64 {
+        let factor: u64 = self.gas_price_factor.into();
+        self.new_scaled_da_gas_price
+            .checked_div(factor)
+            .expect("factor is never 0")
+    }
```

Comment:
We can avoid these expects by using [NonZeroU64](https://doc.rust-lang.org/std/num/type.NonZeroU64.html) (which was recently stabilized in 1.79 🙌), in which case normal division is infallible.

![image](https://github.com/user-attachments/assets/223ee3c8-1ff6-41f7-a093-a9a44ec5ee33)

Reply:
This is hot.

Reply:
Oh I se we already used `NonZeroU64` even. Nice :blush: